File: Chess/src/Rook.java

public class Rook extends Piece{
	private boolean canCastle;
	
	public Rook(String type, Board b, Player p) {
		super(type, b, p);
		this.canCastle=true;
	}
	
	public void move() {
		this.canCastle=false;
	}
	
	public void castle() {
		if(!this.canCastle) {
			return;
		}
	}
	
	public String toString() {
		String t="";
		if(type.equals("Black")) {
			t+="B";
		}
		else {
			t+="W";
		}
		t+="R";
		return t;
	}
}

File: Chess HTML/rook.js
class rook extends piece{
    
    constructor(player,board,x,y){
        super(player,board,x,y,"rook");
        this.hasMoved=false;
        this.capture=false;
    }
    
    move(x,y){
        if(x==this.x){
            if(this.board.checkSpot(x,y)==null || this.board.checkSpot(x,y).player!=this.player){
                if(this.isMoveLegal(x,y,false)){
                    //console.log("Move is legal");
                    return true;
                }
            }
        }
        else if(y==this.y){
            if(this.board.checkSpot(x,y)==null || this.board.checkSpot(x,y).player!=this.player){
                if(this.isMoveLegal(x,y,false)){
                    //console.log("Move is legal");
                    return true;
                }
            }
        }
        return false;
    }
    
}